src/main/java/uk/org/tombolo/importer/ons/CensusImporter.java

```diff
@@ -86,21 +89,22 @@ protected void importDatasource(Datasource datasource, List<String> geographySco
             attributes.add(AttributeUtils.getByProviderAndLabel(attribute.getProvider(), attribute.getLabel()));
         }
 
-        // FIXME: Generalise this beyond LSOA
-        SubjectType lsoa = SubjectTypeUtils.getOrCreate(AbstractONSImporter.PROVIDER,
-                OaImporter.OaType.lsoa.name(), OaImporter.OaType.lsoa.datasourceSpec.getDescription());
+        String subjectTypeFromRecipe = subjectRecipes.get(0).getSubjectType();
```

Comment:
I know that in the examples we don't have any case of this but in the code we support it. we iterate through the subjects in the recipe to write them all in the output file. So here I guess we should do the same in case we have e.g 2 subjects that require lsoa and msoa.

Reply:
yes i thought that about 10 minutes ago and already modified the code